### PR TITLE
Return Book ready for review

### PR DESCRIPTION
**Pull request by pyroib**
submitting for code review

### CODE REVIEW

File: src/ReturnBookControl.java [hunk 1]
```diff
@@ -1,77 +1,88 @@
 public class ReturnBookControl {
+
```
Comment (jazzie-b) [inline src/ReturnBookControl.java]: Too much white space, refer to 4.6.1

File: src/ReturnBookControl.java [hunk 2]
```diff
@@ -1,77 +1,88 @@
...
+    private ReturnBookUI UI;
 
-	private ReturnBookUI ui;
-	private enum CONTROL_STATE { INITIALISED, READY, INSPECTING };
-	private CONTROL_STATE state;
-	
-	private library library;
-	private loan currentLoan;
-	
+    private enum ControlState {
```
Comment (jazzie-b) [inline src/ReturnBookControl.java]: Line breaks are optional but not necessary in enum declarations, refer to 4.8.1

File: src/ReturnBookControl.java [hunk 3]
```diff
@@ -1,77 +1,88 @@
...
-		ui.display(currentLoan.toString());
-		
-		if (currentLoan.isOverDue()) {
-			ui.display(String.format("\nOverdue fine : $%.2f", overDueFine));
-		}
-		ui.setState(ReturnBookUI.UI_STATE.INSPECTING);
-		state = CONTROL_STATE.INSPECTING;		
-	}
+    
+    public void setUI(ReturnBookUI UI) {
```
Comment (jazzie-b) [inline src/ReturnBookControl.java]: Method name doesn't follow required camel back, reffer to 5.2.3

File: src/ReturnBookControl.java [hunk 4]
```diff
@@ -1,77 +1,88 @@
...
+        if (!state.equals(ControlState.INITIALISED)) {
+            throw new RuntimeException("ReturnBookControl: cannot call setUI "
```
Comment (jazzie-b) [inline src/ReturnBookControl.java]: If statement indentation should be 2 spaces not 4 (issue through out file), refer to 4.8.4.1

File: src/ReturnBookControl.java [hunk 5]
```diff
@@ -1,77 +1,88 @@
...
+        state = ControlState.READY;
+    }
+    
+    
+    public void bookScanned(int bookId) {
+        if (!state.equals(ControlState.READY)) {
+            throw new RuntimeException("ReturnBookControl: cannot call "
+                    + "bookScanned except in READY state");
+        }
+        book currentBook = library.Book(bookId);
```
Comment (jazzie-b) [inline src/ReturnBookControl.java]: Unnecessary white space, refer to 4.6.1

File: src/ReturnBookControl.java (hunk 4 above)
Comment (jazzie-b) [inline src/ReturnBookControl.java]: Indentation for if statements/switch statements should be indented 2 not 4, refer to 4.2

**Comment by pyroib** [comment]
As per recommendations from Jim Tulip I have merges the branches ReturnBookControl and ReturnBookUI.

This will be the branch moving forward for my work.

This will require another complete review. Sorry :(

**Pull request closed without merging**